services/src/main/java/org/keycloak/protocol/oidc/endpoints/UserInfoEndpoint.java

```diff
@@ -140,6 +141,8 @@ private Response issueUserInfo(String tokenString) {
             token = verifier.verify().getToken();
         } catch (VerificationException e) {
             event.error(Errors.INVALID_TOKEN);
+            // See: https://openid.net/specs/openid-connect-core-1_0.html#UserInfoError
+            response.getOutputHeaders().put(HttpHeaders.WWW_AUTHENTICATE, Collections.singletonList(String.format("Bearer realm=\"%s\", error=\"%s\", error_description=\"%s\"", realm.getName(), OAuthErrorException.INVALID_TOKEN, e.getMessage())));
```

Comment:
I'm not sure about pushing `e.getMessage` into `error_description`. Although I think the exception thrown by the verifier does not expose details about the runtime, it may happen.

Maybe just description `Token verification failed` is enough.

Reply:
Okay, I'll adjust the error message for this case.